### ISSUE TITLE
I2C? Or maybe just uart?

### ISSUE THREAD

**Issue by K-4U** [dckiller51/esphome-yeelight-led-screen-light-bar#3]
Hi there,

I've been looking into getting the remote working and was working under the assumption that this remote would be linked through i2c, as already configured by you.

However, with a logic analyzer hooked up, i noticed that i was only getting data, no clock.
Looking further into it i was able to find that the data can be decoded as uart at 4800 baud.

Unfortunately i can't get the rest of the PCB out, something appears to be stuck while pulling the big long board out that (presumably?) holds the remote ic?

How certain are you that the protocol used is i2c? Do you perhaps have a better picture of the board? As i don't have the tools to remove the iron cap.

**Comment by stefankonig** [comment]
I have found the same thing. it appears the PCB designed for i2c, but might later be revised to use UART. however i found it to use a baud rate of 5000. which is a weird number, granted, but the one that worked for me.

You can see what's going on using the following debug config in ESPHome:

```
uart:
  baud_rate: 5000
  rx_pin: GPIO2
  debug:
    direction: RX
    dummy_receiver: true
    after:
      delimiter: "\n"
    sequence:
      - lambda: UARTDebug::log_int(direction, bytes, ',');
```

it seems to repeat every message couple of times for good measure.
This is what i have figured out thus far:

example:

`90,1,2,5,255,0,0,97`

90: seems static
1: a auto incremental number for each unique message
2 : part 1 of the identifier which button/action is pressed
5: part 2 of the identifier which button/action is pressed
255: part 3 of the identifier which button/action is pressed
0: seems static
0: seems static
97: the parity number of all the numbers above it:

Parity calculation:  90+1+2+5+255+0+0 = 353
353 modulo 256 = 97

**Comment by K-4U** [comment]
I've since figured out that it's apparently a protocol called SWS, Single Wire Slave.
It's a properietary protocol from the manufacturer of the chip.

I've found the same thing at 4800 BAUD. It might be that this was just right.
90 is 0x5A, which, according to the manual indicates the start of a message.

I am in the progress of writing a custom component so that i can get the remote working. Or at least the basic on/off functionality of it.
